File: python-introdutorio/avaliacao/PP-P004-PROVA/ListaSalarios.py
from AnaliseDados import AnaliseDados

class ListaSalarios(AnaliseDados):

    def __init__(self):
        super().__init__(float)
        self.__lista = []        

    def entradaDeDados(self):
        n = int(input("Quantos salários deseja incluir na lista? "))
        for i in range(n):
            salario = float(input(f"Informe o {i+1}º salário: "))
            self.__lista.append(salario)
        print("Salário(s) adicionado(s) com sucesso!")

    def mostraMediana(self):
        self.listarEmOrdem()
        n = len(self.__lista)
        if n % 2 == 0:
            mediana = (self.__lista[n//2 - 1] + self.__lista[n//2]) / 2
        else:
            mediana = self.__lista[n//2]
        print(f"Mediana dos salários: {mediana}")

    def mostraMenor(self):
        menor_salario = min(self.__lista)
        print(f"Menor salário: {menor_salario}")

    def mostraMaior(self):
        maior_salario = max(self.__lista)
        print(f"Maior salário: {maior_salario}")
    
    def listarEmOrdem(self):
        self.__lista.sort()
        print("Salários em ordem crescente:", self.__lista)
        
    def reajusteSalarios(self):
        self.__lista = list(map(lambda x: round(x * 1.1, 2), self.__lista))

    def __str__(self):
        return str(self.__lista)

File: modulo-introdutorio/avaliacao-02/residevs-avaliacao-2/ListaSalarios.cpp
#include "include/ListaSalarios.h"

void ListaSalarios::entradaDeDados()
{
    float elemento;
    int qtde;
    bool flag;

    do
    {
        // solicitar quantidade de elementos
        cout << "Quantos elementos vão existir na lista? ";
        cin >> qtde;

        // verificar se o número fornecido é válido, isto é, acima de 0
        flag = isNumeroValido(qtde);

        if (!flag)
        {
            cout << "Aviso: Entre com um número a partir de 1." << endl;
        }
    } while (!flag);

    // solicitar cada elemento
    for (int i = 0; i < qtde; i++)
    {
        cout << "Elemento " + to_string(i + 1) + ": ";
        cin >> elemento;
        //-colocar validacao

        lista.push_back(elemento);
    }
}

void ListaSalarios::mostraMenor()
{
    float menor;
    getValorMinimo(lista, menor);

    cout << "O menor dos salarios é: " << menor << endl;
}

void ListaSalarios::mostraMediana()
{
    int divisao = (lista.size() / 2);
    float mediana;

    if (isPar(divisao))
    {
        mediana = (lista[divisao] + lista[divisao - 1]) / 2;
    }
    else
    {
        mediana = lista[divisao];
    }

    cout << "A mediana da lista de salários é: " << mediana << endl;
}

void ListaSalarios::mostraMaior()
{
    float maior;
    getValorMaximo(lista, maior);

    cout << "O maior dos salarios é: " << maior << endl;
}

void ListaSalarios::ordenarLista()
{
    int tamanho = lista.size();

    for (int i = 1; i < tamanho; i++)
    {
        for (int j = 0; j < tamanho - 1; j++)
        {
            if (lista[j] > lista[j + 1])
            {
                int temp;
                temp = lista[j];
                lista[j] = lista[j + 1];
                lista[j + 1] = temp;
            }
        }
    }
}

void ListaSalarios:: listarEmOrdem()
{
    ordenarLista();
    int tamanho = lista.size();
    cout << "Lista em ordem crescente: " << endl;

    for (int i = 0; i < tamanho; i++)
    {
        cout << lista[i] << endl;
    }
}

void ListaSalarios::listarNPrimeiros()
{
    int tamanho = lista.size();
    int n = 0;

    do
    {
        cout << "Deseja ver os dados até qual índice? (entre 0 - " << tamanho << "):";
        cin >> n;
    } while (n < 0 || n > tamanho);

    cout << "Lista dos " << n << " primeiros: " << endl;
    for (int i = 0; i < n; i++)
    {
        cout << lista[i] << endl;
    }
}